lib/mongo/protocol/msg.rb

```diff
@@ -275,6 +275,28 @@ def bulk_write?
         num_inserts > 1  || num_updates > 1 || num_deletes > 1
       end
 
+      def maybe_add_server_api(client)
+        if client.nil?
+          raise ArgumentError, 'Client must be provided'
+        end
+
+        if server_api = client.options[:server_api]
+          main_document = @main_document.dup
+          if version = server_api[:version]
+            main_document['apiVersion'] = version
+          end
+          unless server_api[:strict].nil?
+            main_document['apiStrict'] = server_api[:strict]
```

Comment:
Are we ok with the user being able to set this to any truthy value?

Reply:
@p-mongo Was for you. The spec expects a boolean value but the user could potentially provide any truthy value here. There's nothing in the spec that I saw about validation here in dynamic languages so I was just asking if we were ok with that potentially erroring if a user supplied, say a string. I'm not sure what the server would do in that case.

Reply:
The server will reject any non-boolean value:
```
MongoDB Enterprise > db.adminCommand({isMaster: 1, apiVersion: "1", apiStrict: "asdf"})
{
	"ok" : 0,
	"errmsg" : "BSON field 'APIParametersFromClient.apiStrict' is the wrong type 'string', expected type 'bool'",
	"code" : 14,
	"codeName" : "TypeMismatch"
}
```

While the spec specifies that the `strict` and `deprecationErrors` options are booleans, it does not specify whether this should trigger a client-side or server-side error. I'm leaning toward client-side errors to better distinguish the wrong configuration from other command errors. Let me know if I should clarify this in the spec.

Reply:
The documentation specifies true and false as the only allowed values. Supplying other values would be contrary to documented usage, hence a user error (unless the server happens to accept other values in the future for some reason in which case it'll be a-ok). If the spec mandates client-side rejection of other values I will be happy to implement that.

Reply:
I'm fine leaving this as-is for the time being.

Reply:
Looks good to me then as well.